Analysis/src/BaseEventAnalyzer.cpp

```diff
@@ -213,18 +213,32 @@ void BaseEventAnalyzer::ProcessDataSource(const SampleDescriptor& sample, const
                 auto btag_weight_provider = eventWeights_HH->GetProviderT<mc_corrections::BTagWeight>(
                     mc_corrections::WeightType::BTag);
 
+
+                    static const std::vector<UncertaintySource> btag_sources = { UncertaintySource::btag_lf,
+                        UncertaintySource::btag_hf, UncertaintySource::btag_hfstats1, UncertaintySource::btag_hfstats2,
+                        UncertaintySource::btag_lfstats1, UncertaintySource::btag_lfstats2,
+                        UncertaintySource::btag_cferr1, UncertaintySource::btag_cferr2,
+                        UncertaintySource::JetFull_Total, UncertaintySource::JetReduced_Total
+                     };
+
+
                 for(const bool iter_fit : {false, true}) {
                     for(const auto wp : btag_wps){
                         const std::pair<DiscriminatorWP, bool> key(wp, iter_fit);
                         btag_weights[key][UncertaintyScale::Central] = static_cast<float>(
                                 btag_weight_provider->Get(*event, wp, iter_fit, unc_source, unc_scale));
                         if(unc_source == UncertaintySource::None && !iter_fit) {
-                            btag_weights[key][UncertaintyScale::Up] = static_cast<float>(
-                                    btag_weight_provider->Get(*event, wp, iter_fit, UncertaintySource::Eff_b,
-                                                              UncertaintyScale::Up));
-                            btag_weights[key][UncertaintyScale::Down] = static_cast<float>(
-                                    btag_weight_provider->Get(*event, wp, iter_fit, UncertaintySource::Eff_b,
-                                                              UncertaintyScale::Down));
+                            for(const auto scale : {UncertaintyScale::Up, UncertaintyScale::Down})
+                                btag_weights[key][scale] = static_cast<float>(btag_weight_provider->Get(*event, wp,
+                                    iter_fit, UncertaintySource::Eff_b, scale));
+                        }
+                        else if(unc_source == UncertaintySource::None && iter_fit) {
+                            for(const auto scale : {UncertaintyScale::Up, UncertaintyScale::Down}) {
+                                for(UncertaintySource unc : btag_sources) {
```

Comment:
- `UncertaintySource::JetFull_Total` and `UncertaintySource::JetReduced_Total` should be applied only when corresponding unc are applied (with and without apply_jes true)
- having `btag_weights` as a map starts to be a bit difficult (and hard to read). We need to add a simple struct to store btag weights.